server/incore-common/src/main/java/edu/illinois/ncsa/incore/common/utils/JsonUtils.java

```diff
@@ -49,4 +60,150 @@ public static String extractValueFromJsonString(String inId, String inJson) {
+
+        double hazard_size_kb = hazard_file_size / 1024;
+        double hazard_size_mb = hazard_size_kb / 1024;
+        double hazard_size_gb = hazard_size_mb / 1024;
+
+        // round values
+        dataset_size_kb = Math.round(dataset_size_kb * 100.0) / 100.0;
+        dataset_size_mb = Math.round(dataset_size_mb * 100.0) / 100.0;
+        dataset_size_gb = Math.round(dataset_size_gb * 100.0) / 100.0;
+        hazard_size_kb = Math.round(hazard_size_kb * 100.0) / 100.0;
+        hazard_size_mb = Math.round(hazard_size_mb * 100.0) / 100.0;
+        hazard_size_gb = Math.round(hazard_size_gb * 100.0) / 100.0;
+
+        String out_dataset_size;
+        String out_hazard_size;
+
+        if (dataset_size_gb >= 1) {
+            out_dataset_size = dataset_size_gb + " GB";
+        } else if (dataset_size_mb >= 1) {
+            out_dataset_size = dataset_size_mb + " MB";
+        } else {
+            out_dataset_size = dataset_size_kb + " KB";
+        }
+
+        if (hazard_size_gb >= 1) {
+            out_hazard_size = hazard_size_gb + " GB";
+        } else if (hazard_size_mb >= 1) {
+            out_hazard_size = hazard_size_mb + " MB";
+        } else {
+            out_hazard_size = hazard_size_kb + " KB";
+        }
+
+        JSONObject outJson = new JSONObject();
+        if (isGroup) {
+            outJson.put("group", username);
+        } else {
+            outJson.put("user", username);
+        }
+        outJson.put("total_number_of_datasets", usage.getDatasets());
+        outJson.put("total_number_of_hazards", usage.getHazards());
+        outJson.put("total_number_of_hazard_datasets", usage.getHazardDatasets());
+        outJson.put("total_number_of_dfr3", usage.getDfr3());
+        outJson.put("total_file_size_of_datasets", out_dataset_size);
+        outJson.put("total_file_size_of_datasets_byte", dataset_file_size);
+        outJson.put("total_file_size_of_hazard_datasets", out_hazard_size);
+        outJson.put("total_file_size_of_hazard_datasets_byte", usage.getHazardDatasetSize());
+
+        return outJson;
+    }
+
+    public static String parseUserName(String userInfo) {
+        String userName = null;
+        try {
+            JSONParser parser = new JSONParser();
+            org.json.simple.JSONObject userInfoJson = (org.json.simple.JSONObject) parser.parse(userInfo);
+            userName = (String) userInfoJson.get("preferred_username");
+
+        } catch (ParseException e) {
+            logger.error("Unable to parse userInfo", e);
+            throw new IncoreHTTPException(Response.Status.BAD_REQUEST, "Unable to parse userInfo");
+        }
+
+        return userName;
+    }
+
+    public static Boolean isLoggedInUserAdmin(String userGroup){
+        Boolean isAdmin = false;
+
+        try {
+            JSONParser parser = new JSONParser();
+            org.json.simple.JSONObject userGroupJson = (org.json.simple.JSONObject) parser.parse(userGroup);
```

Comment:
This method throws an exception and returns 500 response code when the header doesn't contain `x-auth-usergroup` header (userGroup arg is null). It should fail gracefully with an appropriate API response message

Reply:
updated